lib/src/util/dart_type_utilities.dart

```diff
@@ -47,13 +47,33 @@ class DartTypeUtilities {
   static bool hasInheritedMethod(MethodDeclaration node) =>
       lookUpInheritedMethod(node) != null;
 
+  static InterfaceType implementedInterface(
+      DartType type, String interface, String library) {
+    if (type is! InterfaceType) {
+      return null;
+    }
+    bool predicate(InterfaceType i) => isInterface(i, interface, library);
+
+    ClassElement element = type.element;
+    if (predicate(type)) {
+      return type;
+    }
+    if (!element.isSynthetic && type is InterfaceType) {
+      final interfaces = element.allSupertypes.where(predicate);
```

Comment:
This works in the case where the class specifies a non-type parameter as the type of the interface, but I'm concerned that we loose type information if we have something like
```
abstract class SpecialIterator<E> implements Iterator<E> {}
void printItems(SpecialIterator<String> items) ...
```
Asking for the type of `items` will get you `SpecialIterator<String>`, but the element associated with that is `SpecialIterator<E>`. Its supertypes should then include `Iterator<E>`, but `E` won't match `String`.

Unfortunately, there is no `allSupertypes` on `InterfaceType`. Maybe there should be...